packages/kit/src/runtime/client/client.js

```diff
@@ -48,6 +48,19 @@ function update_scroll_positions(index) {
 	scroll_positions[index] = scroll_state();
 }
 
+const fetch = window.fetch;
```

Comment:
It won't matter a ton if these aren't treeshaken fully in prod mode, but would it be better to just declare them as undefined `let`s here (which definitely would get treeshaken) and then assign them inside the `if (import.meta.env.DEV)` below?

Reply:
the later calls to `fetch`...

https://github.com/sveltejs/kit/blob/7556a10afa3333791eaeeeb1d4321d339b382cd6/packages/kit/src/runtime/client/client.js#L547

https://github.com/sveltejs/kit/blob/7556a10afa3333791eaeeeb1d4321d339b382cd6/packages/kit/src/runtime/client/client.js#L645-L647

...would attempt to call an undefined function if the `let` wasn't initialised. This felt like the least worst option but i'm certainly open to alternatives

Reply:
`let original_fetch;` maybe? I dunno. I think that would work without further changes to the rest of the existing code?

I think it's worth stumbling over ourselves a little bit in service of having dev mode code completely treeshaken out in prod mode.

Reply:
if we do `original_fetch`, then those later uses of `fetch` need to use `original_fetch` instead, which means we need that to exist in prod too, no?

Reply:
I haven't looked closely at the code, but I has assumed that we'd want to be using the global `fetch`/`window.fetch` everywhere, so that any calls would be run through our version that warned users about calling it. But it sounds like that's not the case? Do we want all later calls to _not_ trigger the warning? If so, yeah, I agree that what you have now is the least-bad version I can think of (and apologies for the noise!).

Reply:
Yeah, we want the cases in the snippets embedded above to call the original `fetch` and not the intercepted version:

```js
export async function load({ fetch }) {
  // this calls our fetch, which
  //   1. tracks the dependency
  //   2. loads the data from the HTML, if it was serialized and we're hydrating
  //   3. defers to original fetch (should not warn, because the user did everything right)
  const res = await fetch('https://api.example.com/whatever.json');
  // ...
}
```

```js
export async function load() {
  // this calls the intercepted window.fetch, which should print a warning
  const res = await fetch('https://api.example.com/whatever.json');
  // ...
}
```